test/cases/satellite.sh

```diff
@@ -0,0 +1,220 @@
+LoadModule ssl_module /usr/lib64/httpd/modules/mod_ssl.so
+
+TypesConfig mime.types
+
+PidFile logs/httpd.pid
+
+# Default file to serve
+DirectoryIndex index.html
+
+# Errors go to their own log
+ErrorLog logs/error_log
+
+# Access log
+LogFormat "%h %l %u %t \"%r\" %>s %b" common
+CustomLog logs/access_log common
+
+# Never change this block
+<Directory />
+  AllowOverride None
+  Require all denied
+</Directory>
+STOPHERE
+}
+
+# Provision the software under tet.
+/usr/libexec/osbuild-composer-test/provision.sh
+
+greenprint "Installing dependencies"
+sudo dnf install -y openssl httpd mod_ssl
+
+WORKDIR=/tmp/repobug
+mkdir ${WORKDIR}
+cd ${WORKDIR}
+
+MIRROR1_DIR=${WORKDIR}/mirror1/
+MIRROR2_DIR=${WORKDIR}/mirror2/
+
+greenprint "Creating certification authorities"
+mkdir -p "${MIRROR1_DIR}/ca"
+mkdir -p "${MIRROR2_DIR}/ca"
+pushd "${MIRROR1_DIR}/ca"
+generate_certificates
+popd
+pushd "${MIRROR2_DIR}/ca"
+generate_certificates
+popd
+
+greenprint "Initialize httpd configurations"
+pushd "${MIRROR1_DIR}"
+init_httpd_work_dir
+cat >> httpd.conf << STOPHERE
+# Port to Listen on
+Listen 7080
+
+<VirtualHost *:7080>
+   # Just pass all the requests to the real mirror
+   ProxyPass /repo/ https://rpmrepo.osbuild.org/v1/anon/f33/f33-x86_64-fedora-20201108/
+   ProxyPassReverse /repo/ https://rpmrepo.osbuild.org/v1/anon/f33/f33-x86_64-fedora-20201108/
+   # The real mirror redirects to this URL, so proxy this one as well, otherwise
+   # it won't work with the self-signed client certificates
+   ProxyPass /aws/ https://rpmci.s3.us-east-2.amazonaws.com/
+   ProxyPassReverse /aws/ https://rpmci.s3.us-east-2.amazonaws.com/
+   # But turn on SSL
+   SSLEngine on
+   SSLProxyEngine on
+   SSLCertificateFile ca/server.crt
+   SSLCertificateKeyFile ca/server.key
+   # And require the client to authenticate using a certificate issued by our custom CA
+   SSLVerifyClient require
+   SSLVerifyDepth 1
+   SSLCACertificateFile ca/ca.crt
+</VirtualHost>
+STOPHERE
+popd
+pushd "${MIRROR2_DIR}"
+init_httpd_work_dir
+cat >> httpd.conf << STOPHERE
+# Port to Listen on
+Listen 7081
+
+<VirtualHost *:7081>
+   # Just pass all the requests to the real mirror
+   ProxyPass /repo/ https://rpmrepo.osbuild.org/v1/anon/f33/f33-x86_64-fedora-modular-20201108/
+   ProxyPassReverse /repo/ https://rpmrepo.osbuild.org/v1/anon/f33/f33-x86_64-fedora-modular-20201108/
+   # The real mirror redirects to this URL, so proxy this one as well, otherwise
+   # it won't work with the self-signed client certificates
+   ProxyPass /aws/ https://rpmci.s3.us-east-2.amazonaws.com/
+   ProxyPassReverse /aws/ https://rpmci.s3.us-east-2.amazonaws.com/
+   # But turn on SSL
+   SSLEngine on
+   SSLProxyEngine on
+   SSLCertificateFile ca/server.crt
+   SSLCertificateKeyFile ca/server.key
+   # And require the client to authenticate using a certificate issued by our custom CA
+   SSLVerifyClient require
+   SSLVerifyDepth 1
+   SSLCACertificateFile ca/ca.crt
+</VirtualHost>
+STOPHERE
```

Comment:
I think you should be able to drop this under `/etc/httpd/conf.d/` instead of overriding httpd.conf